### PR TITLE
Update contributor docs

### PR DESCRIPTION
**Pull request by rosteen**
Some comments on the PRs/issues resulting from the recent conference brought to my attention that the developer docs are very out of date. Hopefully this update will be more useful to new contributors.

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -64,25 +64,31 @@ Ready to contribute? Here's how to set up :ref:`specutils <specutils>` for local
...
 
     $ git checkout -b name-of-your-bugfix-or-feature
 
    Now you can make your changes locally.
 
-5. When you're done making changes, check that your changes pass flake8 and the tests, including testing other Python versions with tox::
+5. When you're done making changes, check that your changes pass flake8 and the tests::
+
+    $ tox -e codestyle
+    $ pytest
```
Comment (kelle) [inline docs/contributing.rst:84]: So, `pytest` didn't work for me. I had to run `tox` with the python-version specific config file specified.  If `pytest` _should_ work, I'm ok leaving this, but just heads up.

**Review by kelle — COMMENTED** [review]
I left some comments/have some questions.

File: docs/contributing.rst [hunk 2]
```diff
@@ -64,25 +64,31 @@ Ready to contribute? Here's how to set up :ref:`specutils <specutils>` for local
...
-3. Install your local copy into a virtualenv. Assuming you have virtualenvwrapper installed, this is how you set up your fork for local development::
+3. Install your local copy, preferably into some sort of virtual environment using your
+   preferred environment manager. For example, using conda::
 
-    $ mkvirtualenv specutils
+    $ conda create --name specutils-dev python pip
+    $ conda activate specutils-dev
     $ cd specutils/
-    $ python setup.py develop
+    $ pip install -e .[test]
```
[the remark this answers is not in the document]
Comment (rosteen) [reply docs/contributing.rst:73]: Adding `[test]` after the dot also installs optional dependencies needed to run the tests locally.

File: docs/contributing.rst (hunk 1 above)
Comment (rosteen) [reply docs/contributing.rst:84]: Running `pytest` should work, and if it doesn't currently should be fixed. Are there further problems after the PR you opened the other day?

**Review by kelle — APPROVED** [review]
Next time I make changes, I'll test all this out, but for the moment, this looks like great progress! Thanks!

**Pull request merged**